techmatters/flex-plugins
# Chi 1435: Refactor contact saving logic redux promise - Part 2

Primary reviewer: @stephenhand 

## Description


- **Feature refactor**: Refactor contact save logic into redux-promise middleware - part 2 (`connectToCase` and `submitContactForm`).
The ticket is the second part of the contact save logic refactor. It handles the refactoring of network calls for contact to be handled in our Redux layer rather than the component layer. This is the first PR for the `connectToCase` and `submitContactForm` services layer method.

- **Bug fix**: It also fixes the `Save and End Case` error.
- TODO: Implement units for the newly implemented actions and reducers. 


### Checklist

### Related Issues
Fixes [CHI-1435](https://tech-matters.atlassian.net/browse/CHI-1435)

### Verification steps

- Create an offline contact
- Fill in all the necessary details
- Click on "Save Contact"
- Verify that the contact was saved successfully
- Repeat steps 1 and 2
- Click on "Add Contact to New Case"
- Click on "Save and End"
- Verify that contact was created and successfully added to the case

[CHI-1435]: https://tech-matters.atlassian.net/browse/CHI-1435?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
